lib/tapioca/compilers/requires_compiler.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+# typed: true
+
+module Tapioca
+  module Compilers
+    class RequiresCompiler
+      extend(T::Sig)
+
+      sig { params(sorbet_path: String).void }
+      def initialize(sorbet_path)
+        @sorbet_path = sorbet_path
+      end
+
+      sig { returns(String) }
+      def compile
+        files = ruby_files(@sorbet_path)
+
+        files.map do |file|
+          collect_requires(file)
+        end.flatten.sort.uniq.reject do |req|
+          req = T.cast(req, String)
+          name = /['"]([^'"]+)['"]/.match(req)
+          next unless name
+          project_require?(files, T.must(name[1]))
+        end.join
+      end
+
+      private
+
+      sig { params(sorbet_path: String).returns(T::Array[String]) }
+      def ruby_files(sorbet_path)
+        parser = SorbetConfigParser.new(sorbet_path)
+        return [] unless parser.config?
+        parser.ruby_files
+      end
+
+      sig { params(file_path: String).returns(T::Enumerable[String]) }
+      def collect_requires(file_path)
+        File.read(file_path).lines.select do |line|
+          line =~ /^require /
+        end
+      end
+
+      sig { params(files: T::Enumerable[String], name: String).returns(T::Boolean) }
+      def project_require?(files, name)
+        files.any? do |file|
+          file =~ /#{name}/
+        end
+      end
+
+      class SorbetConfigParser
+        extend T::Sig
+
+        sig { params(sorbet_path: String).void }
+        def initialize(sorbet_path)
+          @sorbet_path = sorbet_path
+          @dirs = T.let([], T::Array[String])
+          @ignored = T.let([], T::Array[Regexp])
+          read_config if config?
+        end
+
+        sig { returns(T::Array[String]) }
+        def ruby_files
+          @dirs.map do |dir|
+            Dir.glob("#{dir}/**/*.rb", File::FNM_EXTGLOB).map do |file|
```

Comment:
`flat_map` and return `[]` instead of `nil` if ignore and you don't need `compact` anymore.